### PR TITLE
raft: Export Progress.IsPaused

### PR DESCRIPTION
**Pull request by bdarnell**
CockroachDB would like to use this method for monitoring.

cc @xiang90 @petermattis

### CODE REVIEW

File: raft/progress.go [hunk 1]
```diff
@@ -155,8 +155,11 @@ func (pr *Progress) maybeDecrTo(rejected, last uint64) bool {
 func (pr *Progress) pause()  { pr.Paused = true }
 func (pr *Progress) resume() { pr.Paused = false }
 
-// isPaused returns whether progress stops sending message.
-func (pr *Progress) isPaused() bool {
+// IsPaused returns whether sending log entries to this node has been
+// paused. A node may be paused because it has rejected recent
+// MsgApps, is currently waiting for a snapshot, or has reached the
+// MaxInflightMsgs limit.
+func (pr *Progress) IsPaused() bool {
```
Comment (xiang90) [inline raft/progress.go]: the access of pr.Paused is not protect by lock. I am not sure if this access is safe.

Comment (bdarnell) [inline raft/progress.go]: `getStatus` makes a copy of each `Progress` struct, so it should be fine.

Comment (xiang90) [inline raft/progress.go]: Ah. OK. Can you check status is the only way external pkg to access Progress? If yes, then lgtm.

Comment (bdarnell) [inline raft/progress.go]: Yes, I think it is. In any case, if there were other ways to access the `Progress` struct then all of its fields would have the same synchronization problem.

**Comment by xiang90** [comment]
LGTM

**Pull request merged**